TM1py/Utils/Utils.py

```diff
@@ -33,11 +33,11 @@ def format_url(url, *args: str, **kwargs: str) -> str:
     :param args: arguments to placeholders
     :return:
     """
-    args = [arg.replace("'", "''") if isinstance(arg, str) else arg
+    args = [arg.replace("'", "''").replace('%', '%25') if isinstance(arg, str) else arg
```

Comment:
Could you give me an example here?  I tried creating a subset with name `test %`, `te % st`, `% test` and `t% est` everything worked.

```Python
subset = Subset(subset_name='t% est', dimension_name='test %', hierarchy_name='test %',
                expression="{TM1FILTERBYLEVEL({TM1SUBSETALL ([test %])}, 0)}")
tm1.subsets.create(subset)
```

Reply:
The issue we were having was not the subset name but the elements names contain a variety of characters such as $, #, >, %, so when querying using MDX  using the code below, we received errors

    mdx2 = """
    SELECT
    NON EMPTY {{TM1FILTERBYLEVEL( TM1SUBSETALL( [Material APAC] ), 0)}} ON ROWS,
    {{[Time Month].[2018-M02],[Time Month].[2018-M03],
    [Time Month].[2018-M04],[Time Month].[2018-M05],[Time Month].[2018-M06],
    [Time Month].[2018-M07],[Time Month].[2018-M08],[Time Month].[2018-M09],[Time Month].[2018-M10],
    [Time Month].[2018-M11],[Time Month].[2018-M12],[Time Month].[2019-M01],[Time Month].[2019-M02],
    [Time Month].[2019-M03],[Time Month].[2019-M04],[Time Month].[2019-M05],[Time Month].[2019-M06],
    [Time Month].[2019-M07],[Time Month].[2019-M08],[Time Month].[2019-M09],[Time Month].[2019-M10],
    [Time Month].[2019-M11],[Time Month].[2019-M12],[Time Month].[2020-M01],[Time Month].[2020-M02],
    [Time Month].[2020-M03],[Time Month].[2020-M04],[Time Month].[2020-M05]}} ON COLUMNS
    FROM [Allocation Assumptions APAC]
    WHERE ([VERSION].[ACT], [Customer APAC].[Total Customer APAC],[Allocation Account APAC].[Undefined GL Account],
    [Allocation Assumptions Drivers APAC Measure].[Drivers],[Purchasing Org].[Total Purchasing Org],
    [Profit Center].[Total Profit Center], [Allocation Region APAC].[Asia Pacific] )"""

    df2 = tm1.cubes.cells.execute_mdx_dataframe(mdx2)
    used_assumption_materials = set(df2['Material APAC'].unique())

    unused_materials = set(all_materials) - used_staging_materials - used_assumption_materials

    dimension_name = "Material APAC"
    subset_name = "Unused Materials APAC"
    elements = set(unused_materials)
    #expression = "{" + ",".join("[Material APAC].[" + element + "]" for element in elements) + "}"
    s = Subset(dimension_name=dimension_name, subset_name=subset_name, alias='', elements=elements)
    tm1.dimensions.subsets.create(subset=s, private=False)

When executing, even with expression instead of using elements the below errors were thrown but again the symbols are in the elements names, not the subset name.

![image](https://user-images.githubusercontent.com/8358847/87901212-55bb2300-ca0b-11ea-9d0a-4ecdcd9b5633.png)

Reply:
Thanks for giving the sample, I will look into it. But this merge will cause you problems.

Reply:
Did this MR fix your issue?

Reply:
I have not been able to test it yet.  I have to have the client pip the update, which they did , but the servers are down so I have not been able to test yet.  Probably not until tomorrow morning.

Reply:
If possible, please give me the element names which are causing the issue. I will anyway try to check for $, #, > and %.

Reply:
All the below elements are working perfectly (they are all UTF-8 characters). Can you copy/paste that element `787KXD` here?

![image](https://user-images.githubusercontent.com/60260968/87908434-ff3dfb00-ca83-11ea-9c84-fac18f480f9a.png)

Reply:
Closing the conversation:

RC: Element names with `%AD` were creating the issue, as the subset names were not getting encoded.
#292 is fixing it by encoding `%` to `%25`.